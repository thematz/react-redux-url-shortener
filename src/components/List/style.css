.source-url {
    word-break: break-word;
}

.source-url:hover {
    text-decoration: underline;
}