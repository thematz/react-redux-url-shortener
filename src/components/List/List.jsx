// src/components/List/index.jsx

import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import TimeAgo from 'react-timeago'

import { fetchList } from '../../actions'
import './style.css'

class List extends Component {
    componentWillMount() {
        this.props.fetchList()
    }
    render() {
        return (
            <section className="section">
                <div className="container">
                    <div className="columns">
                        <div className="column is-8 is-offset-2">
                            {this.props.list.map(item => {
                                return (
                                    <div className="card" key={item.id}>
                                        <div className="card-content">
                                            <p>
                                                <a href={"/" + item.id} target="_blank" className="has-text-primary short-url">{item.id}</a>
                                                <span className="is-pulled-right is-size-7">
                                                    <TimeAgo date={item.creationDate} />
                                                </span>
                                            </p>
                                            <p>
                                                <a href={item.url} className="is-small has-text-grey is-size-7 has-text-weight-semibold source-url">{item.url}</a>
                                            </p>
                                        </div>
                                    </div>
                                )
                            })
                            }
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}

const mapStateToProps = state => ({
    list: state.list
})
const mapDispatchToProps = dispatch => bindActionCreators({
    fetchList
}, dispatch)

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(List)
