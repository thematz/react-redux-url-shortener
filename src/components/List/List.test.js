import React from 'react'
import List from './List'
import store from '../../store'

import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
Enzyme.configure({ adapter: new Adapter() })

it('renders List correctly', () => {
    const wrapper = shallow(
        <List store={store} />,
    ).dive()
    expect(toJson(wrapper)).toMatchSnapshot()
})