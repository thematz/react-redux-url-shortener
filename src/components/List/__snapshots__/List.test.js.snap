// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List correctly 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <div
      className="columns"
    >
      <div
        className="column is-8 is-offset-2"
      />
    </div>
  </div>
</section>
`;
